packages/devtools_app/lib/src/performance/performance_controller.dart

```diff
@@ -206,9 +216,28 @@ class PerformanceController
     data.selectedFrame = frame;
     _selectedFrameNotifier.value = frame;
 
-    await selectTimelineEvent(frame.uiEventFlow);
+    // We do not need to pull the CPU profile because we will pull the profile
+    // for the entire frame. The order of selecting the timeline event and
+    // pulling the CPU profile for the frame (directly below) matters here.
+    // If the selected timeline event is null, the event details section will
+    // not show the progress bar while we are processing the CPU profile.
+    await selectTimelineEvent(frame.uiEventFlow, pullCpuProfile: false);
```

Comment:
What if we added a CPU Profile model/databse that tracked all the time ranges of profiling data we have already cached. That way we wouldn't need to track whether to set pullCpuProfile to true or not. If the time range is already cached. That would also enable supporting offline mode for the cpu profiler data in the future as the offline data could include the database of cpu profile samples that were cached.